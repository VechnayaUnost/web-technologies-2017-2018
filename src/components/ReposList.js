import React from 'react';

import Repo from './Repo';

class ReposList extends React.Component {
    render() {
        return (
            <ul>
                {
                    this.props.repositories.map((repo) =>
                        <Repo repo={repo}/>
                    )
                }
            </ul>
        )
    }
}

export default ReposList;