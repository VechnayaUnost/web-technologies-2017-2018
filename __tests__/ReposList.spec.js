import React from 'react';
import ReposList from '../src/components/ReposList';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('ReposList component test', () => {
    it('ReposList component rendered succesfully', () => {
        const props = {
            repositories: [
                {
                    name: 'AvatarSkeleton',
                    description: 'Creates Avatar skeleton with some preinstalled options',
                    stargazers_count: 321,
                    language: 'C#',
                },
                {
                    name: 'AvatarMountConnection',
                    description: 'Avatar mount connection is easier with this tool',
                    stargazers_count: 543,
                    language: 'Java',
                },
            ]
        }

        const reposList = mount(
            <ReposList {...props} />
        );
        expect(mountToJson(reposList)).toMatchSnapshot();
        expect(reposList.find('Repo').length).toBe(2);
    });
});