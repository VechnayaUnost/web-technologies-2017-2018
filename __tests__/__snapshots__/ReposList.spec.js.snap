// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposList component test ReposList component rendered succesfully 1`] = `
<ReposList
  repositories={
    Array [
      Object {
        "description": "Creates Avatar skeleton with some preinstalled options",
        "language": "C#",
        "name": "AvatarSkeleton",
        "stargazers_count": 321,
      },
      Object {
        "description": "Avatar mount connection is easier with this tool",
        "language": "Java",
        "name": "AvatarMountConnection",
        "stargazers_count": 543,
      },
    ]
  }
>
  <ul>
    <Repo
      repo={
        Object {
          "description": "Creates Avatar skeleton with some preinstalled options",
          "language": "C#",
          "name": "AvatarSkeleton",
          "stargazers_count": 321,
        }
      }
    >
      <li
        className="repo"
      >
        <div
          className="repo-description"
        >
          <h3>
            AvatarSkeleton
          </h3>
          <p>
            Creates Avatar skeleton with some preinstalled options
          </p>
        </div>
        <div
          className="repo-stars-lang"
        >
          <span
            className="repo-lang"
          >
            Language: 
            C#
          </span>
          <span
            className="repo-stars"
          >
            Stars: 
            321
          </span>
        </div>
      </li>
    </Repo>
    <Repo
      repo={
        Object {
          "description": "Avatar mount connection is easier with this tool",
          "language": "Java",
          "name": "AvatarMountConnection",
          "stargazers_count": 543,
        }
      }
    >
      <li
        className="repo"
      >
        <div
          className="repo-description"
        >
          <h3>
            AvatarMountConnection
          </h3>
          <p>
            Avatar mount connection is easier with this tool
          </p>
        </div>
        <div
          className="repo-stars-lang"
        >
          <span
            className="repo-lang"
          >
            Language: 
            Java
          </span>
          <span
            className="repo-stars"
          >
            Stars: 
            543
          </span>
        </div>
      </li>
    </Repo>
  </ul>
</ReposList>
`;
